[CN-532] Add CronHotBackup documentation (#51)
hazelcast/hazelcast-platform-operator-docs: docs/modules/ROOT/pages/backup-restore.adoc

kubectl apply -f ./hot-backup.yaml
----

=== Scheduling Backups with CronHotBackup Custom Resource

You can schedule regular backups, using the `.spec.schedule` and `.spec.hotBackupTemplate` fields of the `CronHotBackup` resource. You can find more detailed information about the `CronHotBackup` resource in the xref:api-ref.adoc#cronhotbackupspec[API Reference] page.

. Create the `CronHotBackup` resource with `schedule` and `hotBackupTemplate` fields.
+
[source,yaml,subs="attributes+"]
----
include::ROOT:example$/cron-hot-backup.yaml[]
----
+
`.spec.schedule` field takes a valid cron expression. For example:

> +

Author: Sure, I am updating it

kubectl apply -f ./hot-backup.yaml
----

=== Scheduling Local Backups

You can schedule regular backups, using the `schedule` and `hotBackupTemplate` fields of the `CronHotBackup` resource. For more information about the `CronHotBackup` resource, see the xref:api-ref.adoc#cronhotbackupspec[API Reference].

. Create the `CronHotBackup` resource and add values for the `schedule` and `hotBackupTemplate` fields. For example:
+
[source,yaml,subs="attributes+"]
----
include::ROOT:example$/cron-hot-backup.yaml[]
----
+
The `schedule` field takes a valid cron expression. For example, you can configure the following scheduled backups:
+
[cols="1,1"]
|===
| ```30 10 * * *```
| At 10:30 AM every day

| ```0, 0, 1,15,25 * *```
| On 1st, 15th, and 25th of each month at midnight

| ```@monthly```
| On the first day of each month at midnight
|===

+
For a full list of support expressions, see the link:https://pkg.go.dev/github.com/robfig/cron/v3@v3.0.1#hdr-CRON_Expression_Format[library documentation].

. Apply the resource:
+
[source,bash]
----
kubectl apply -f ./cron-hot-backup.yaml
----

=== Checking the Status of a Local Backup

> To check the status of a backup, run the following command:

Reviewer: ```suggestion
To check the status of a local backup, run the following command:
```

+
[source,bash]
----
kubectl apply -f ./cron-hot-backup.yaml
----

> === Checking the Status of a Backup

Reviewer: ```suggestion
=== Checking the Status of a Local Backup
```
Also, should we say.... "To check the status of a local backup, run the following command:". Also do we need to provide an example to show how to check the status of a local scheduled backup/is it the same? Can you check the status of a specific scheduled backup, i.e. by filename?

kubectl apply -f ./hot-backup-agent.yaml
----

=== Scheduling External Backups with CronHotBackup Custom Resource

You can schedule external backups, using the `.spec.schedule` and `.spec.hotBackupTemplate` fields of the `CronHotBackup` resource. You can find more detailed information about the `CronHotBackup` resource in the xref:api-ref.adoc#cronhotbackupspec[API Reference] page.

. Create the `CronHotBackup` resource with `schedule` and `hotBackupTemplate` fields.
+
[source,yaml,subs="attributes+"]
----
include::ROOT:example$/cron-hot-backup-agent.yaml[]
----
+
`.spec.schedule` field takes a valid cron expression. For example:

> +

Reviewer: For the table below, just some suggestions for slight rewordings.

At 10.30 AM every day

On 1st, 15th, and 25th of each month at midnight

On the first day of each month at midnight

For a full list of support expressions, see the library documentation.

Author: I am updating it

=== Scheduling External Backups

You can schedule regular backups to an external storage bucket, using the `schedule` and `hotBackupTemplate` fields of the `CronHotBackup` resource. For more information about the `CronHotBackup` resource, see the xref:api-ref.adoc#cronhotbackupspec[API Reference].

. Create the `CronHotBackup` resource and add values for the `schedule` and `hotBackupTemplate` fields. For example:
+
[source,yaml,subs="attributes+"]
----
include::ROOT:example$/cron-hot-backup-agent.yaml[]
----
+
The `schedule` field takes a valid cron expression. For example, you can configure the following scheduled backups:
+
[cols="1,1"]
|===
| ```30 10 * * *```
| At 10:30 AM every day

| ```0, 0, 1,15,25 * *```
| On 1st, 15th, and 25th of each month at midnight

| ```@monthly```
| On the first day of each month at midnight
|===

+
For a full list of support expressions, see the link:https://pkg.go.dev/github.com/robfig/cron/v3@v3.0.1#hdr-CRON_Expression_Format[library documentation].

. Apply the resource:
+
[source,bash]
----
kubectl apply -f ./cron-hot-backup-agent.yaml
----

=== Checking the Status of an External Backup

> To check the status of a backup, run the following command:

Reviewer: ```suggestion
To check the status of an external backup, run the following command:
```

+
[source,bash]
----
kubectl apply -f ./cron-hot-backup-agent.yaml
----

> === Checking the Status of a Backup

Reviewer: ```suggestion
=== Checking the Status of an External Backup
```
Can you check the status of a specific scheduled backup, i.e. by filename?

Also, could we say: "To check the status of an external backup, run the following command:"
